images/README.md

```diff
@@ -0,0 +1,110 @@
+# Building Images
+
+This directory contains tooling for building base images for use as nodes in Kubernetes Clusters. [Packer](https://www.packer.io) is used for building these images.
+
+## Prerequisites
+
+### Prerequisites for all images
+
+- [Packer](https://www.packer.io/docs/installation.html)
+- [Ansible](http://docs.ansible.com/ansible/latest/intro_installation.html) version >= 2.4.0.0
+
+### Prerequisites for Amazon Web Services
+
+- An AWS account
+- The AWS CLI installed and configured
+
+## Building Images
+
+### Build Variables
+
+The following variables can be overriden when building images using the `-var` option when calling `packer build`:
+
+| Variable | Default | Description |
+|----------|---------|-------------|
+| build_version | unset | A unique build version for the image |
+| kubernetes_version | 1.11.2-00 | Kubernetes Version to install |
+| kubernetes_cni_version | 0.6.0-00 | CNI Version to install |
+
+For example, to build all images for use with Kubernetes 1.11.2 for build version 1:
+
+```sh
+packer build -var kubernetes_version=1.11.2-00 -var build_version=1
+```
+
+There are additional variables that may be set that affect the behavior of specific builds or packer post-processors. `packer inspect packer.json` will list all available variables and their default values.
+
+### Limiting Images to Build
+
+If packer build is run without specifying which images to build, then it will attempt to build all configured images. `packer inspect packer.json` will list the configured builders. The `--only` option can be specified when running `packer build` to limit the images built.
+
+For example, to build only the Ubuntu image:
+
+```sh
+packer build -var build_version=`git rev-parse HEAD` --only=ami-ubuntu packer.json
+```
+
+### Required Permissions to Build the AWS AMIs
+
+The [Packer documentation for the Amazon AMI builder](https://www.packer.io/docs/builders/amazon.html) supplies a suggested set of minimum permissions. However, Wardroom has been successfully tested with the following IAM permissions:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2:Describe*",
+                "ec2:TerminateInstances",
+                "ec2:StartInstances",
+                "ec2:CreateSnapshot",
+                "ec2:CreateImage",
+                "ec2:RunInstances",
+                "ec2:StopInstances",
+                "ec2:CreateKeyPair",
+                "ec2:DeleteKeyPair",
+                "ec2:CreateSecurityGroup",
+                "ec2:DeleteSecurityGroup",
+                "ec2:AuthorizeSecurityGroupIngress",
+                "ec2:CreateTags",
+                "ec2:DeleteVolume"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
```

Comment:
Have you been able to test using these limited permissions? I'm wondering if the use of `ami_groups` or `ami_regions` adds additional requirements.

Reply:
I've updated the permissions to match the official packer suggested ones. Which include permissions to modify images attributes, needed for ami_groups to function. The `ami_regions` should be good too.